### ISSUE TITLE
[Bug Report] setup_hugepages.py stops after permission error

### ISSUE THREAD

**Issue by changh95** [tenstorrent/tt-metal#11825]
**Describe the bug**

When setting up environment for TT-Buda, users are expected to run `setup_hugepages.py`.
This should run flawlessly, but an exception arises during the script and there's no way to fix it other than fixing the Python script.

There's a file called 'demote' in '/sys/kernel//mm/hugepages/hugepages-1048576kB/'.
From what I remember, this file does not have read permission.
This causes the python script to fail to read the item size, even with sudo.

I got a fixed version of python script - PR will follow after this issue.

**To Reproduce**
(Following the instructions from TT-Buda-demos repo)
1. Download driver, flash firmware, and other stuffs.
2. `wget https://raw.githubusercontent.com/tenstorrent/tt-metal/main/infra/machine_setup/scripts/setup_hugepages.py` to download python script
3. Run `sudo -E python3 setup_hugepages.py first_pass` and encounter error

**Expected behavior**

The error is not supposed to happen, and a reboot sequence should be suggested.

**Please complete the following environment information:**
- OS: Ubuntu 24.04
- Version of software `6e16b48ee3066ffccbf8411b3c5155b09971f8fc`

**Comment by milank94** [comment]
@changh95 that method of installing hugepages on Ubuntu 24.04 won't work.

Can you try these updated steps: https://github.com/tenstorrent/tt-buda-demos/blob/mkordic/rc_20240830/first_5_steps/1_install_tt_buda.md#setup-hugepages

**Comment by changh95** [comment]
@milank94 
Strange - it worked on my clean Ubuntu 24.04.
I now have a dev environment fully set-up, so I can't wipe it and make a clean 24.04 environment again.

Do you have any suggestions how to do this?

**Comment by milank94** [comment]
@changh95 I see, sorry I misunderstood the first message.

So you encountered and error with the script, made a correction, and then will push an update?

Can you share details about the error?

As I mentioned, we have updated the hugepages setup process. I believe this should mitigate these issues in the future.

**Comment by changh95** [comment]
@milank94 

In the `setup_hugepages.py` file, there's `read_kernel_hugepages_info()` function, which basically tries to estimate the page size in bytes. This function tries to read the file `demote` in `/sys/kernel//mm/hugepages/hugepages-1048576kB/`, but this demote file has no read permission. So even with sudo-ing the python script, the python script just fails with permission error.

> @milank94 Strange - it worked on my clean Ubuntu 24.04, and that's how I got to make this PR. I now have a dev environment fully set-up, so I can't wipe it and make a clean 24.04 environment again.
> 
> Do you have any suggestions how to do this?

And when I wrote this comment, I thought this page was the PR itself, not the issue 😅 .
My PR on this fixed the error (https://github.com/tenstorrent/tt-metal/pull/11826), and was merged 2 weeks ago.

If the new updated steps don't work on Ubuntu 24.04, I have no way of testing it out since I already established a full dev environment on my personal PC with Ubuntu 24.04. 

If the new updated steps completely avoids this error, I say this issue can be closed :)
